Remotion/Core/UnitTests/Context/SafeContextBoundaryTests.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Moq;
+using NUnit.Framework;
+using Remotion.Context;
+
+namespace Remotion.UnitTests.Context
+{
+  [TestFixture]
+  public class SafeContextBoundaryTests
+  {
+    [Test]
+    public void Dispose ()
+    {
+      var boundaryStrategyMock = new Mock<ISafeContextBoundaryStrategy>();
+      var state = new object();
+
+      var safeContextBoundary = new SafeContextBoundary(boundaryStrategyMock.Object, state);
+      safeContextBoundary.Dispose();
+
+      boundaryStrategyMock.Verify(e => e.RestorePreviousSafeContext(state));
+    }
+
+    [Test]
+    public void Dispose_WithNullBoundaryStrategy_ThrowsNothing ()
+    {
+      var safeContextBoundary = new SafeContextBoundary();
```

Comment:
I wonder if we should use `default` here instead. C# 10 allows default ctors for structs with an implementation so this code might mask an issue in the future.